core/mock_test.go

```diff
@@ -328,50 +329,121 @@ func newMockCluster(
 		}
 
 		// Create a new instance of the IBFT node
-		i := NewIBFT(logger, backend, transport)
+		nodes[index] = NewIBFT(logger, backend, transport)
 
-		// Instantiate quit channels for node routines
-		quitChannels[index] = make(chan struct{})
-		messageHandlersQuit[index] = make(chan struct{})
-
-		nodes[index] = i
+		// Instantiate context for the nodes
+		ctx, cancelFn := context.WithCancel(context.Background())
+		nodeCtxs[index] = &mockNodeContext{
```

Comment:
Since we ruled out this wasn't the cause of a failing test, feel free to make it a value (rather than a pointer)